File: app/src/main/java/com/customerglu/customerglu/MyFirebaseMessagingService.java
package com.customerglu.customerglu;

import android.os.Build;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;


import org.json.JSONObject;

public class MyFirebaseMessagingService extends FirebaseMessagingService {


    @RequiresApi(api = Build.VERSION_CODES.M)
    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);
        JSONObject data = null;
        JSONObject json = new JSONObject(remoteMessage.getData());

                MainActivity.customerGlu.displayCustomerGluNotification(this,json, R.drawable.customerglu,0.5);

        }

    @Override
    public void onNewToken(@NonNull String s) {
        super.onNewToken(s);

    }

    @Override
    public void onSendError(@NonNull String s, @NonNull Exception e) {
        super.onSendError(s, e);
        Toast.makeText(getApplicationContext(), "my ex"+e, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/customerglu/MyFirebaseMessagingService.kt
package com.example.customerglu

import android.annotation.SuppressLint
import android.util.Log
import com.customerglu.sdk.CustomerGlu
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import org.json.JSONObject


@SuppressLint("MissingFirebaseInstanceTokenRefresh")
class MyFirebaseMessagingService: FirebaseMessagingService() {

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        Log.d("FirebaseMessage", "Message data payload: ${remoteMessage.data}")
//        val body: String = remoteMessage.data.toString()
//        val json = JSONObject(body)
//        Log.d("json", json.toString())
        val str = JSONObject(remoteMessage.data as Map<*, *>?)
        Log.d("Cg",str.toString())
//        if (str.has("type")) {
            CustomerGlu.getInstance()
                .displayCustomerGluNotification(applicationContext, str, R.drawable.bags);
   //     }


//        if (remoteMessage.data.isNotEmpty()) {
//            Log.d("FirebaseMessage", "Message data payload: ${remoteMessage.data}")
//
//        }
    }
}